### DISCUSSION TITLE
The 2 biggest issues with npm and node

### DISCUSSION THREAD

**Discussion by excitedbox** [npm/feedback#90] [General]
The 2 biggest issues with npm and node in general are complexity and repetition/redundancy. These could both be solved quite easily with the addition of a gui.

The complexity and learning curve is a huge time waster for everyone and a giant source of frustration. Getting everything to run smoothly takes months especially for those with less experience. This is mostly caused by the lack of GUIs and forcing everyone to use the command line and config files. Constantly having to look up syntax and commands wastes millions of hours a year in the industry and is a multi billion $ problem. This is worsened by the constant issues of solving issues with wrong paths or syntax usage.

The few gui options out there are buggy and lack 90% of the most important features but already save tons of time in finding and installing packages.

The second part is redundancy. Node, Electron, Python, Jupyter, etc each start their own module repository on the system and in some cases updating one of these starts up another one. Unless you are a power user and can't easily consolidate these and configure them all to use the same folders. This could also be solved through the use of a gui and better standardization across all these applications.

I don't think I really need to add anymore. We all know the pain of having to use the command line and how long it took us all to get even close to something workable. I don't think that in the year 2020 we should still be relying on these outdated systems from the 50 years ago. There is a reason we don't use DOS anymore and abstract away as much as possible. This is just another method of abstraction. Sadly this same problem persists in many other ecosystems which is why we all have at least 3 python installations cluttering our SSDs with millions of packages.

I propose a proper that it is time to finally put an end to this.

**Comment by MylesBorins** [comment]
Hey @excitedbox thanks for your enthusiastic post.

Have you had a chance to play with [node-red](https://nodered.org/)? It is a fantastic visual programming approach to node.js.

While I can appreciate the frustration you are feeling there are many developers, myself included, who run software on systems that don't have screens or UIs (e.g. servers in the cloud). As a maintainer of Node.js I don't think that shipping a visual tool is within scope for the project which is primarily run by volunteers.

As for npm writing a visually based tool for managing packages is not something that will be in scope for our work either. The JavaScript ecosystem is text based, and the degree of change you are asking for is something that would require the entire ecosystem to change how they work.
